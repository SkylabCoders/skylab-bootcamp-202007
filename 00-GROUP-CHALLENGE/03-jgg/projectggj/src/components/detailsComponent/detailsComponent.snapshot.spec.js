import renderer from 'react-test-renderer';
import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import DetailsComponent from './detailsComponent';

function renderDetails(arg) {
    const props = {
        match: {
            params: {
                name: 'Vegeta'
            }
        }
    };

    return renderer.create(
        <Router>
            <DetailsComponent {...props} />
        </Router>
    );
}

describe('DetailsComponent snapShot', () => {
    let DetailTree;
    let instance;
    let component;
    let text;

    beforeEach(async () => {
        DetailTree = renderDetails();
        DetailTree.update();
    });

    it('should macth', () => {
        let DetailsComponentTree = renderDetails();

        expect(DetailsComponentTree).toMatchSnapshot();
    });
    it('should display name of chracter', async () => {
        DetailTree = renderDetails();

        instance = DetailTree.root;
        component = instance.findAllByType('a')[0];

        text = component.children[0];

        expect(text).toEqual('Vegeta');
    });
    it('should display name of chracter', async () => {
        DetailTree = renderDetails();

        instance = DetailTree.root;
        component = instance.findAllByProps({ className: 'details-desc' })[0];
        text = component.findByType('span').children;

        expect(...text).toEqual('Saiyan');
    });
});