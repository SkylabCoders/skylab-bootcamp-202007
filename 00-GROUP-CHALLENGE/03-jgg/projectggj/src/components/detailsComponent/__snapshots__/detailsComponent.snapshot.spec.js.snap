// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsComponent snapShot should macth 1`] = `
<div
  className="details-holder"
>
  <div
    className="flex-item"
  >
    <div
      className="row"
    >
      <div
        className="detail-container flex-item"
      >
        <div
          className="card hovercard"
        >
          <div
            className="cardheader"
          />
          <div
            className="avatar"
          >
            <img
              alt=""
              src="https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcSSrgcSH0bbTsOigHcPRNUbC1OGldwFLDx3kg&usqp=CAU"
            />
          </div>
          <div
            className="info"
          >
            <div
              className="title"
            >
              <a
                href="/"
                onClick={[Function]}
                target="_blank"
              >
                Vegeta
              </a>
            </div>
            <div
              className="details-desc-content"
            >
              <div
                className="details-desc"
              >
                RACE: 
                <span>
                  Saiyan
                </span>
              </div>
              <div
                className="details-desc"
              >
                STATUS: 
                <span>
                  Alive
                </span>
              </div>
              <div
                className="details-desc"
              >
                PLANET: 
                <span>
                  Vegeta
                </span>
              </div>
              <div
                className="details-desc"
              >
                GENDER: 
                <span>
                  Male
                </span>
              </div>
              <div
                className="details-desc"
              >
                SERIE: 
                <span>
                  Z
                </span>
              </div>
            </div>
            <div>
              <a
                href="/game/Vegeta"
                onClick={[Function]}
              >
                <button
                  className="row figthlink btn"
                >
                   FIGTH!
                </button>
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
